### ISSUE TITLE
Date: Site date formats may crash the editor

### ISSUE THREAD

**Issue by sirreal** [WordPress/gutenberg#27514]
**Describe the bug**

The block editor may crash with the following error when attempting to edit a post. The editor displays `The editor has encountered an unexpected error` and this error will be printed in the console:

```
react-dom.min.de439aae.js:125 RangeError: Invalid time zone specified: -02
    at new DateTimeFormat (<anonymous>)
    at index.js?ver=3e8c0d6780df04db4fcebbd77d1875f9:1
    at me (index.js?ver=3e8c0d6780df04db4fcebbd77d1875f9:1)
    at Object.z (index.js?ver=3e8c0d6780df04db4fcebbd77d1875f9:1)
    at index.js?ver=3e8c0d6780df04db4fcebbd77d1875f9:1
    at Array.reduce (<anonymous>)
    at Fe (index.js?ver=3e8c0d6780df04db4fcebbd77d1875f9:1)
    at Object.e (index.js?ver=3e8c0d6780df04db4fcebbd77d1875f9:1)
    at Ke (index.js?ver=3e8c0d6780df04db4fcebbd77d1875f9:1)
    at ut (index.js?ver=3e8c0d6780df04db4fcebbd77d1875f9:1)
```

**To reproduce**

1. In _Settings > General_, set the date format to `e`. Any string including a plain `e` seems to be sufficent to trigger the error, for example users may have `j de F de Y` (which was likely intended as `j \d\e F \d\e Y`).
1. It may be necessary to use a UTC offset date and not a region. This was reproduced with `UTC-2` set as the site timezone.
1. Publish a new post.
1. Attempt to open the post for editing (the error also appears to happen on publish).
1. Observe the error.

**Expected behavior**

The editor should just work fine.

**Editor version (please complete the following information):**
- WordPress version: 5.5.3
- If the Gutenberg plugin is installed, which version is it? 9.5.0

**Comment by sirreal** [comment]
Users should be able to work around the problem by not using a plain `e` in their date format setting.

**Comment by ockham** [comment]
Potentially fixed by #27398.

**Comment by david-szabo97** [comment]
Fixed by #27550
